installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -22,68 +23,99 @@ enum Shell
+            SHA256 sha = SHA256.Create();
+            FileStream fInfo = File.OpenRead(zipPath);
+            string zipHash = BitConverter.ToString(sha.ComputeHash(fInfo)).Replace("-", string.Empty).ToLower();
+            if (zipHash != info.sha256v2)
+            {
+                string msg = string.Format("SHA256 checksum did not match, expected: {0} actual: {1}", info.sha256v2, zipHash.ToString());
+                session.Log(msg);
+                ActiveState.RollbarHelper.Report(msg);
+                return ActionResult.Failure;
+            }
 
-            string output;
-            ActionResult result = ActiveState.Command.Run(session, "powershell", installCmd, out output);
-            if (result.Equals(ActionResult.UserExit))
+            Status.ProgressBar.StatusMessage(session, "Extracting State Tool executable...");
+            try
             {
-                // Catch cancel and return
-                ActiveState.RollbarHelper.Report("Installation exited prematurely due to UserExit (probably safe to ignore)");
-                return result;
+                ZipFile.ExtractToDirectory(zipPath, tempDir);
             }
-            else if (result.Equals(ActionResult.Failure))
+            catch (Exception e)
             {
-                Record record = new Record();
-                var errorOutput = FormatErrorOutput(output);
-                record.FormatString = String.Format("state tool installation failed with error:\n{0}", errorOutput);
-
-                MessageResult msgRes = session.Message(InstallMessage.Error | (InstallMessage)MessageBoxButtons.OK, record);
-                ActiveState.RollbarHelper.Report(record.FormatString);
-                return result;
+                string msg = string.Format("Could not extract State Tool, encountered exception: {0)", e);
+                session.Log(msg);
+                ActiveState.RollbarHelper.Report(msg);
+                return ActionResult.Failure;
             }
-            Status.ProgressBar.Increment(session, 1);
 
-            stateToolPath = Path.Combine(installPath, "state.exe");
-            if ( ! File.Exists(stateToolPath))
+            stateToolPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "ActiveState", "bin", "state.exe");
+            try
             {
-                ActiveState.RollbarHelper.Report(String.Format("State tool installed without errors but its filePath does not exist.  install.ps1 output: {0}", FormatErrorOutput(output)));
+                File.Move(Path.Combine(tempDir, "windows-amd64.exe"), stateToolPath);
             }
-            return result;
+            catch (Exception e)
+            {
+                string msg = string.Format("Could not move State Tool executable, encountered exception: {0}", e);
+                session.Log(msg);
+                ActiveState.RollbarHelper.Report(msg);
+                return ActionResult.Failure;
+            }
```

Comment:
This is missing adding the bin path to the environment PATH. Otherwise State tool still won't be usable.